tools/icqsol_add_texture/icqsol_add_texture.py

```diff
@@ -0,0 +1,55 @@
+#!/usr/bin/env python
+import argparse
+import shutil
+from pkg_resources import resource_filename
+
+import icqsol_utils
+from icqsol.shapes.icqShapeManager import ShapeManager
+
+# Parse Command Line.
+parser = argparse.ArgumentParser()
+parser.add_argument('--input', dest='input', help='Shape dataset selected from history')
+parser.add_argument('--input_file_format_and_type', dest='input_file_format_and_type', help='Input file format and type')
+parser.add_argument('--input_dataset_type', dest='input_dataset_type', help='Input dataset_type')
+parser.add_argument('--max_edge_length', dest='max_edge_length', type=float, default='0', help='Maximum edge length')
+parser.add_argument('--texture', dest='texture', default='stone', help='Select texture (currently stone, wood, or checkerboard)')
+parser.add_argument('--output', dest='output', help='Output dataset')
+parser.add_argument('--output_vtk_type', dest='output_vtk_type', help='Output VTK type')
+
+args = parser.parse_args()
+
+input_format, input_file_type = icqsol_utils.get_format_and_type(args.input_file_format_and_type)
+tmp_dir = icqsol_utils.get_temp_dir()
+
+# Instantiate a ShapeManager for loading the input.
+if input_format == icqsol_utils.VTK:
+    shape_mgr = ShapeManager(file_format=input_format, vtk_dataset_type=args.input_dataset_type)
+else:
+    shape_mgr = ShapeManager(file_format=input_format)
+
+# Get the vtk polydata from the input dataset.
+vtk_poly_data = shape_mgr.loadAsVtkPolyData(args.input)
+
+# Add texture. We use setuptools, which installs the package as a python "egg", 
+# essentially a zip file containing the modules and data files. Must use resource_filename
+# to access the data files.
+texture_name_to_filename = {
+    'stone': resource_filename('icqsol', 'textures/220px-COnglomerate-sandstone_layers_Nerriga.jpg'),
+    'wood': resource_filename('icqsol', 'textures/Swietenia_macrophylla_wood.jpg'),
+    'checkerboard': resource_filename('icqsol', 'textures/checkerboard.png'),
+}
+
```

Comment:
@gregvonkuster We could but then the user would need to upload png or jpeg files. (ShapeManager's addTexture method takes a file name).